src/ZlibSharp/ZlibSharp/MemoryZlib.cs

```diff
@@ -8,264 +8,101 @@ namespace ZlibSharp;
-        {
-            // the compression failed because of a support failure.
-            throw new NotPackableException("Compression Failed.", ex);
-        }
-        catch (IOException ex) when (ex is not NotPackableException)
-        {
-            throw new NotPackableException("Compression Failed.", ex);
-        }
+        return Compress(Source.AsSpan(), Dest.AsSpan(), CompressionLevel);
     }
-
-    /// <summary>
-    /// Compresses data using an specific compression level and outputs an adler32 hash with the data.
-    /// </summary>
-    /// <param name="inData">The original input data.</param>
-    /// <param name="level">The compression level to use.</param>
-    /// <exception cref="NotPackableException">
-    /// Thrown when the internal compression stream errors in any way.
-    /// </exception>
-    /// <returns>
-    /// A <see cref="ValueTuple"/> containing the compressed data, as well as the adler32 hash of that data.
-    /// </returns>
-    public static (byte[] OutData, uint Adler32) CompressHash(byte[] inData, ZlibCompressionLevel level)
+    
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static uint Compress(string SourcePath, byte[] Dest, ZlibCompressionLevel CompressionLevel = ZlibCompressionLevel.DefaultCompression)
     {
-        using var outMemoryStream = new MemoryStream();
-        var result = CompressHash(inData, outMemoryStream, level);
-        return (outMemoryStream.ToArray(), result);
+        var Source = File.ReadAllBytes(SourcePath);
+        
+        return Compress(Source, Dest, CompressionLevel);
     }
-
-    /// <summary>
-    /// Compresses a file using an specific compression level and outputs an adler32 hash with the data.
-    /// </summary>
-    /// <param name="path">The file to compress.</param>
-    /// <param name="level">The compression level to use.</param>
-    /// <exception cref="NotPackableException">
-    /// Thrown when the internal compression stream errors in any way.
-    /// </exception>
-    /// <returns>
-    /// A <see cref="ValueTuple"/> containing the compressed data, as well as the adler32 hash of that data.
-    /// </returns>
-    public static (byte[] OutData, uint Adler32) CompressHash(string path, ZlibCompressionLevel level)
-        => CompressHash(File.ReadAllBytes(path), level);
-
-    /// <summary>
-    /// Decompresses data.
-    /// </summary>
-    /// <param name="inData">The compressed input data.</param>
-    /// <param name="outStream">The decompressed output data.</param>
-    /// <exception cref="NotUnpackableException">
-    /// Thrown when the internal decompression stream errors in any way.
-    /// </exception>
-    public static void Decompress(byte[] inData, Stream outStream)
+    
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static uint Compress(Span<byte> Source, Span<byte> Dest, ZlibCompressionLevel CompressionLevel = ZlibCompressionLevel.DefaultCompression)

```

Comment:
After thinking about this perhaps the source spans could be implicitly converted to ``ReadOnlySpan``s and nuke the byte[] overloads so users do not get too big of a breakage.

Reply:
You can't write to a readonlyspan, though

Reply:
I think an implicit cast to Span<byte> is possible, though. Imma try to refactor the API

Reply:
Just realized I can't get rid of the .AsSpan() calls, since it would cause a recursive call into the byte[] API